api/src/pcapi/core/fraud/api.py

```diff
@@ -270,7 +248,7 @@ def duplicate_id_piece_number_fraud_item(user: users_models.User, id_piece_numbe
 
 
 def format_id_piece_number(id_piece_number: str) -> str:
-    return id_piece_number.replace(" ", "")
+    return re.sub(r"[\W^_]", "", id_piece_number.upper())
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Il faudra vérifier ce que ça donne sur l'existant en prod

i.e. exécuter ça sur la base de prod et voir si on a des diffs
Si non -> enregistrer les changements
Si oui -> Voir au cas par cas ?